File: two-pointers/11-container-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0;
        int r = height.size() - 1;
        int ans = 0;
        int water;
        while (l < r){
            water = (r - l) * min(height[l], height[r]);
            ans = max(water, ans);
            if (height[l] < height[r]){
                l += 1;
            } else {
                r -= 1;
            }    
        }
        return ans;
    }
};

File: two-pointers/11-container-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height) - 1
        res = 0
        while l < r:
            heightL = height[l]
            heightR = height[r]
            vol = min(heightL, heightR) * (r - l)
            res = max(res, vol)            
            if heightL > heightR:
                r -= 1
            else:
                l += 1
        return res